src/vivarium_nih_us_cvd/components/effects.py

```diff
@@ -108,3 +109,108 @@ def register_target_modifier(self, builder: Builder) -> None:
+            "high_ldl_cholesterol",
+            "high_fasting_plasma_glucose",
+        ],
+        "acute_myocardial_infarction": [
+            "high_systolic_blood_pressure",
+            "high_ldl_cholesterol",
+            "high_fasting_plasma_glucose",
+        ],
+        "post_myocardial_infarction_to_acute_myocardial_infarction": [
+            "high_systolic_blood_pressure",
+            "high_ldl_cholesterol",
+            "high_fasting_plasma_glucose",
+        ],
+    },
+    "high_fasting_plasma_glucose": {
+        "acute_ischemic_stroke": [
+            "high_ldl_cholesterol",
+        ],
+        "chronic_ischemic_stroke_to_acute_ischemic_stroke": [
+            "high_ldl_cholesterol",
+        ],
+        "acute_myocardial_infarction": [
+            "high_ldl_cholesterol",
+        ],
+        "post_myocardial_infarction_to_acute_myocardial_infarction": [
+            "high_ldl_cholesterol",
+        ],
+    },
+}
+
+
+class MediatedRiskEffect(RiskEffect):
+    """Applies mediation to risk effects"""
+
+    def setup(self, builder):
+        super().setup(builder)
+        # Register unadjusted RR pipelines by passing target=1s to the super's target_modifier
+        self.unadjusted_rr = builder.value.register_value_producer(
+            f"unadjusted_rr_{self.risk.name}_on_{self.target.name}",
+            source=lambda idx: self.target_modifier(idx, pd.Series(1.0, index=idx)),
+        )
+        self.mediators = MEDIATOR_NAMES.get(self.risk.name, {}).get(self.target.name, [])
+        self.unadjusted_mediator_rr = {
+            mediator: builder.value.get_value(
+                f"unadjusted_rr_{mediator}_on_{self.target.name}"
+            )
+            for mediator in self.mediators
+        }
+        # Register the mediation target modifier
+        self.mediated_target_modifier = self.get_mediated_target_modifier(builder)
+        self.register_mediated_target_modifier(builder)
+
+    def register_target_modifier(self, builder: Builder) -> None:
+        """We do not want to register the super's target modifier, so we override it here"""
+        pass
+
+    def get_mediated_target_modifier(
+        self, builder: Builder
+    ) -> Callable[[pd.Index, pd.Series], pd.Series]:
+        mediation_factors = builder.data.load("risk.cause.mediation_factors")
+        mediation_factors = mediation_factors.loc[
+            (mediation_factors["risk_name"] == self.risk.name)
+            & (mediation_factors["affected_entity"] == self.target.name)
+        ]
+
+        def adjust_target(index: pd.Index, target: pd.Series) -> pd.Series:
+            unadjusted_rr = self.unadjusted_rr(index)
+            scaling_factor = pd.Series(1, index=index)
+            for mediator in self.mediators:
+                unadjusted_mediator_rr = self.unadjusted_mediator_rr[mediator](index)
+                not_tmrel_idx = index[
+                    (unadjusted_mediator_rr != 1.0) & (unadjusted_rr != 1.0)
```

Comment:
Don't we only care if the mediator RR is 1? That's where we get the division by 0. The math still works out fine if the mediator is not TMREL, but the unadjusted risk is TMREL.

That said, it is more efficient to check for both as you are doing, because the adjustment is guaranteed to do nothing if there is an unadjusted RR of 1. I think it'd be helpful to add a comment explaining why we're checking for each of the TMRELs here.

Reply:
We had a bit of a discussion about this (https://ihme.slack.com/archives/C01N6LFMN3W/p1695338124205609) and Syl requested that we don't adjust if either RR is 1.